support-frontend/test/selenium/util/PostDeployTestUser.scala

```diff
@@ -4,86 +4,32 @@ import java.time.Duration.ofDays
 
 import com.gu.identity.testing.usernames.TestUsernames
 
-import okhttp3.{OkHttpClient, Request, RequestBody, MediaType, Response}
-import io.circe.generic.semiauto.deriveDecoder
-import io.circe.{Json, Decoder, DecodingFailure}
-import io.circe.parser.parse
-
-import org.slf4j.LoggerFactory
-
 trait TestUser {
   val username: String
 }
 
-class PostDeployTestUserContribs(driverConfig: DriverConfig) extends TestUser {
+class PostDeployTestUser(driverConfig: DriverConfig, bypassRecaptchaCookies: Option[List[IdapiCookie]] = None) extends TestUser {
 
   private val testUsers = TestUsernames(
     com.gu.identity.testing.usernames.Encoder.withSecret(Config.testUsersSecret),
     recency = ofDays(2)
   )
 
-  private def addTestUserCookies(testUsername: String) = {
-    driverConfig.addCookie(name = "pre-signin-test-user", value = testUsername)
-    driverConfig.addCookie(name = "_test_username", value = testUsername, domain = Some(Config.guardianDomain))
-    driverConfig.addCookie(name = "_post_deploy_user", value = "true") // This enables the tests to use the mocked payment services
-  }
-
-  val username = testUsers.generate()
-  addTestUserCookies(username)
-}
-
-class PostDeployTestUserSubs(driverConfig: DriverConfig) extends TestUser {
-
   private def addTestUserCookies(testUsername: String) = {
     driverConfig.addCookie(name = "pre-signin-test-user", value = testUsername)
     driverConfig.addCookie(name = "_test_username", value = testUsername, domain = Some(Config.guardianDomain))
     driverConfig.addCookie(name = "_post_deploy_user", value = "true") // This enables the tests to use the mocked payment services
     driverConfig.addCookie(name = "GU_TK", value = "1.1") //To avoid consent banner, which messes with selenium
-    IdapiRequest.decodeCookies.foreach { cookie =>
-      driverConfig.addCookie(name = cookie.key, value = cookie.value)
+    bypassRecaptchaCookies match {
+      case None => 
+      case Some(cookies) => {
+        cookies.foreach {cookie =>
+          driverConfig.addCookie(name = cookie.key, value = cookie.value)
+        }
+      }
```

Comment:
you can avoid the pattern match by flattening the Option and List together
```suggestion
    bypassRecaptchaCookies.toList.flatten.foreach { cookie =>
      driverConfig.addCookie(name = cookie.key, value = cookie.value)
    }
```

Reply:
I see!

Reply:
Added